test/CMakeLists.txt

```diff
@@ -17,4 +17,4 @@ add_subdirectory(old-tests)
 
 add_subdirectory(install)
 
-configure_file(${CMAKE_SOURCE_DIR}/test/CTestCustom.cmake ${CMAKE_BINARY_DIR})
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/test/CTestCustom.cmake ${CMAKE_CURRENT_BINARY_DIR})
```

Comment:
Since I'm not sure about the binary dir changes, I reverted those. The one change that I'm certain about is using PROJECT_SOURCE_DIR instead of CMAKE_SOURCE_DIR, so that's what I'm using.